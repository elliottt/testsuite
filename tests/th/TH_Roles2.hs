{-# LANGUAGE TemplateHaskell, RoleAnnotations, PolyKinds #-}

module TH_Roles2 where

import Language.Haskell.TH

$( return [DataD [] (mkName "T") [KindedRoledTV (mkName "a") (VarT (mkName "k")) Representational] [] []] )

