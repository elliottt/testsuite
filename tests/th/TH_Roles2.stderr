TYPE SIGNATURES
TYPE CONSTRUCTORS
  T :: forall (k :: BOX). k -> *
  data T (k::BOX)@N (a::k)@R
      No C type associated
      RecFlag NonRecursive, Not promotable
      =
      FamilyInstance: none
COERCION AXIOMS
Dependent modules: []
Dependent packages: [array-0.4.0.2, base, containers-0.5.3.1,
                     deepseq-1.3.0.2, ghc-prim, integer-gmp, pretty-1.1.1.0,
                     template-haskell]

==================== Typechecker ====================

